Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,220 @@
+        private var horizontalAccuracy: Double = -1.0
+        
+        ///  This value tracks whether the source is GPS or BLE.
+        private var source: String = ""
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelText: String = ""
+        
+        /// A indoors location data source based on sensor data, including but not
+        /// limited to radio, GPS, motion sensors.
+        @Published private(set) var indoorsLocationDataSource: IndoorsLocationDataSource?
+        
+        /// The map's location display.
+        @Published private(set) var locationDisplay = LocationDisplay()
+        
+        /// Represents loading state of indoors data, blocks interaction until loaded. 
+        @Published var isLoading = false
+        
+        /// The measurement formatter for sensor accuracy.
+        private let measurementFormatter: MeasurementFormatter = {
+            let formatter = MeasurementFormatter()
+            formatter.unitStyle = .short
+            formatter.unitOptions = .providedUnit
+            return formatter
+        }()
+        
+        /// Kicks off the logic for displaying the indoors position.
+        /// - Parameter dataSourceType: The data model type to use when displaying indoor position.
+        func loadAndDisplayIndoorData(for dataSourceType: DataSourceType) async throws {
+            try await setIndoorDatasource(for: dataSourceType)
+            try await startLocationDisplay()
+            try await updateLocation()
+        }
+        
+        /// A function that attempts to load an indoor definition attached to the map
+        /// and returns a boolean value based whether it is loaded.
+        /// - Parameter map: The map that contains the IndoorDefinition.
+        /// - Returns: A boolean value for whether the IndoorDefinition is loaded.
+        private func loadAndCheckForIndoorDefinition(map: Map) async throws -> Bool {
+            guard let indoorPositioningDefinition = map.indoorPositioningDefinition else { return false }
+            if indoorPositioningDefinition.loadStatus != .loaded {
+                try await indoorPositioningDefinition.load()
+                return indoorPositioningDefinition.loadStatus == .loaded
+            }
+            return true
+        }
+        
+        /// Sets the indoor datasource on the location display depending on
+        /// whether the map contains an IndoorDefinition.
+        /// - Parameter map: The map which is checked for an indoor definition.
+        private func setIndoorDatasource(for type: DataSourceType) async throws {
+            await indoorsLocationDataSource?.stop()
+            indoorsLocationDataSource = nil
+            try await map.floorManager?.load()
+            switch type {
+            case .indoorDefinition:
+                if try await loadAndCheckForIndoorDefinition(map: map),
```

Comment:
@yo1995 Given that we are selecting the mode, what should be the behavior if indoorsDefinition is selected but not present in the map?